asmcli/asmcli.sh

```diff
@@ -2657,19 +2668,21 @@ EOF
 
 print_config() {
   local MANAGED; MANAGED="$(context_get-option "MANAGED")"
-  local CUSTOM_OVERLAY; CUSTOM_OVERLAY="$(context_get-option "CUSTOM_OVERLAY")"
 
   if [[ "${MANAGED}" -eq 1 ]]; then
     cat "${MANAGED_MANIFEST}"
     return
   fi
 
   PARAMS="-f ${OPERATOR_MANIFEST}"
-  while read -d ',' -r yaml_file; do
-    PARAMS="${PARAMS} -f ${yaml_file} "
+  while read -r yaml_file; do
+    if [[ -n ${yaml_file} ]]; then
```

Comment:
Why do we need to check for empty variables here? Can we do it when we're appending to the list instead of every time we're reading it?

If that's not possible, please double quote the "${yaml_file}" here.

Reply:
It will output empty string if the list is empty. I could update the `context_list-istio-yamls` to output nothing if empty?

Reply:
Ah, I should have caught that in the last review. In the `list-istio-yamls` function use `-j` as the flag to `jq` instead of `-r`. `-j` is the same except it suppresses the trailing newline character.

Reply:
Sounds good! Learning everyday 😄

Reply:
Discussed offline. Decision is to use for loop instead of heredoc to avoid the empty line.